.github/workflows/build.yml

```diff
@@ -0,0 +1,134 @@
+name: Build cht-core and test against node versions
+
+on: [push, pull_request]
+
+env:
+  COUCH_URL: http://admin:pass@localhost:5984/medic-test
+  COUCH_NODE_NAME: nonode@nohost
+  BUILDS_SERVER: _couch/builds_testing
+  STAGING_SERVER: _couch/builds
+  UPLOAD_URL: ${{ secrets.UPLOAD_URL }}
+
+
+jobs:
+  build:
+    name: Compile the app 
+    runs-on: ubuntu-18.04
+  
+    strategy:
+      matrix:
+        node-version: [12.x]
+
+    steps: 
+    # - name: Login to Docker Hub
+    #   uses: docker/login-action@v1
+    #   with:
+    #     username: ${{ secrets.DOCKERHUB_USERNAME }}
+    #     password: ${{ secrets.DOCKERHUB_TOKEN }}
+    - name: Set UPLOARD_URL var
+      run: |
+        if [[ -z "$UPLOAD_URL" ]]; then
+          echo "UPLOAD_URL=https://staging.dev.medicmobile.org" >> $GITHUB_ENV
+          echo "BUILDS_SERVER=_couch/builds_external" >> $GITHUB_ENV
+        fi
+    - name: Get branch name
+      uses: nelonoel/branch-name@v1.0.1
+    - name: Set Travis Vars
+      run: |
+        echo "TRAVIS_BUILD_NUMBER=$GITHUB_RUN_ID" >> $GITHUB_ENV
+        echo "TRAVIS_BRANCH=$BRANCH_NAME" >> $GITHUB_ENV
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - name: Couch Start
```

Comment:
I don't think you need CouchDB for the compile step. 
I _think_ that, in the `travis.yml` script, the "main" `before_script` section (this one https://github.com/medic/cht-core/blob/master/.travis.yml#L46) runs only if the job itself doesn't have a `before_script` defined. 

The `build` job _has_ a `before_script` here: https://github.com/medic/cht-core/blob/master/.travis.yml#L79

This means that the "main" `before_script` only runs for e2e and performance jobs, and not for the build job. 

Same applies for the `after_script`. 

So, I think the compile part here can be stripped down quite a bit.

Reply:
That would be awesome to remove all of that. The reason I left it is because there is a script that configures couch as part of the compile. Seems like that might not be necessary either but I'll look a bit deeper into the grunt ci-compile to see if it is needed.

Reply:
You could be right! Now that I think about it, maybe the api migration tests are run against couch!

Reply:
Yep, compile needs couch to run api migration integration tests :(